examples/Thermostat/Device/ThermoBot.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) ZeroC, Inc.
+
+using IceRpc;
+using IceRpc.Features;
+using IceRpc.Slice;
+using Igloo;
+using System.Runtime.CompilerServices;
+
+namespace ThermostatDevice;
+
+/// <summary>Implements all the logic of our thermostat device, without any actual electronics or a real AC unit.
+/// </summary>
+internal class ThermoBot : Service, IThermoControlService
+{
+    internal Task ReadCompleted => _readTcs.Task;
+
+    private Stage _cooling;
+
+    // Protects all read-write fields.
+    private readonly object _mutex = new();
+
+    // Used to signal when the server has stopped reading.
+    private readonly TaskCompletionSource _readTcs = new(TaskCreationOptions.RunContinuationsAsynchronously);
+
+    private float _setPoint = 75.0F;
```

Comment:
Total coincidence, but it's kind of nice how looks and feels like `75°F` == `75.0F`